OneCommunityGlobal/HighestGoodNetworkApp
# Xiao F Create way for owner to edit & update ChatGPT prompt text

# Description
![CleanShot 2023-11-10 at 11 19 59](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/43768723/fea012f3-77fd-4a53-9902-f52daa2bee9b)


## Related PRS (if any):
To test this front-end PR you need to checkout the [#603](https://github.com/OneCommunityGlobal/HGNRest/pull/603) back-end PR.

## Main changes explained:
- Update file `src/components/WeeklySummary/CurrentPromptModal.jsx` to get and update prompt with a fallback prompt if error occurs
- Create and update multiple files for reducer function of get and update prompt

## How to test:
1. Check into current branch
2. Do `npm install` and `npm run start:local` to run this PR locally
3. Clearing cache/site data
4. Log in as owner user
5. Go to dashboard -> Summary -> View and Copy Current AI Prompt -> Edit prompt -> Save
6. Check if prompt is updated
7. Check if 'Copy Prompt' works as intended
8. Log in as roles other than owner
9. Go to dashboard -> Summary -> View and Copy Current AI Prompt -> Make sure 'Edit' does not appear

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/43768723/69ddb11d-4889-4fe4-b841-2af83f120c47

## Note:
N/A
